### PR TITLE
dialects: (affine) use maps in affine.for bounds, and add some helpers

### PR DESCRIPTION
**Pull request by superlopuh**
@Groverkss

### CODE REVIEW

**Review by adutilleul — APPROVED** [review]

**Review by compor — APPROVED** [review]
LGTM

File: xdsl/ir/affine/affine_map.py [hunk 1]
```diff
@@ -15,6 +15,18 @@ class AffineMap:
     num_symbols: int
     results: list[AffineExpr]
 
+    @staticmethod
+    def from_constant(value: int) -> AffineMap:
+        return AffineMap(0, 0, [AffineExpr.constant(value)])
```
Comment (Groverkss) [inline xdsl/ir/affine/affine_map.py:20]: Can we call this constant_map?

File: xdsl/ir/affine/affine_map.py [hunk 2]
```diff
@@ -15,6 +15,18 @@ class AffineMap:
...
+
+    @staticmethod
+    def identity(rank: int) -> AffineMap:
+        return AffineMap(rank, 0, [AffineExpr.dimension(dim) for dim in range(rank)])
+
+    @staticmethod
+    def empty() -> AffineMap:
+        return AffineMap(0, 0, [])
+
```
Comment (Groverkss) [inline xdsl/ir/affine/affine_map.py:29]: Can you make the empty method with dims = 0, syms = 0?

File: xdsl/ir/affine/affine_map.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/ir/affine/affine_map.py:20]: How about just `constant`?

Comment (Groverkss) [reply xdsl/ir/affine/affine_map.py:20]: I prefer `constant_map`. It's also what mlir uses.

File: xdsl/ir/affine/affine_map.py (hunk 2 above)
Comment (superlopuh) [reply xdsl/ir/affine/affine_map.py:29]: you mean explicitly specifying the arguments?

Comment (Groverkss) [reply xdsl/ir/affine/affine_map.py:29]: No, I meant the api. But it's fine, I can extend this later.

**Review by Groverkss — APPROVED** [review]

**Pull request merged**